### PR TITLE
switched to `StandardScaler` from deprecated `normalize`

### LINKED ISSUE #7097: `FutureWarning` issued by `scikit-learn` in `opflow`
### Information

- **Qiskit Terra version**: 0.18+
- **Python version**: any
- **Operating system**: any

### What is the current behavior?

With the new(ish) `scikit-learn` version 1.0, there is a `FutureWarning` issued when using the `normalize` parameter of `LinearModel`, which `opflow` does when calculating gradients. This doesn't fail our tests (though perhaps it should), because `FutureWarning` doesn't count as a type of `DeprecationWarning`.

### Steps to reproduce the problem

Run the test suite, especially `test.python.opflow.test_gradients.TestGradients.test_natural_gradient`.

### What is the expected behavior?

No future warning.

### Suggested solutions

Fix `opflow` to handle fitting with `StandardScaler` rather than the now-deprecated `normalize` parameter.  This has existed for several minor releases of `scikit-learn`, so should be safe for us to do.

### PR DESCRIPTION
**Pull request by prakharb10**
### Summary
Closes #7097 

### Details and comments
Removed the deprecated param and added `StandardScaler`

### CODE REVIEW

File: releasenotes/notes/switched-to-StandardScaler-43d24a7918e96c14.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+upgrade:
+  - |
+    The :meth:`.NaturalGradient._ridge` and :meth:`.NaturalGradient._lasso` methods
+    now use the :class:`~sklearn.preprocessing.StandardScaler` to scale the data
+    before fitting the model if the ``normalize`` parameter is set to ``True``.
```
Comment (Cryoris) [inline releasenotes/notes/switched-to-StandardScaler-43d24a7918e96c14.yaml:6]:
```suggestion
fixes:
  - |
    Fix deprecation warnings in :class:`.NaturalGradient`, which now uses the
    now use :class:`~sklearn.preprocessing.StandardScaler` to scale the data
    before fitting the model if the ``normalize`` parameter is set to ``True``.
```

**Review by Cryoris — APPROVED** [review]
LGTM!

File: releasenotes/notes/switched-to-StandardScaler-43d24a7918e96c14.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
-upgrade:
+fixes:
   - |
-    The :meth:`.NaturalGradient._ridge` and :meth:`.NaturalGradient._lasso` methods
-    now use the :class:`~sklearn.preprocessing.StandardScaler` to scale the data
+    Fix deprecation warnings in :class:`.NaturalGradient`, which now uses the
+    now use :class:`~sklearn.preprocessing.StandardScaler` to scale the data
```
Comment (prakharb10) [inline releasenotes/notes/switched-to-StandardScaler-43d24a7918e96c14.yaml:5]: Thank you, @Cryoris. I believe there's a repetition in the reno here - `now uses the` , `now use`.

File: releasenotes/notes/switched-to-StandardScaler-43d24a7918e96c14.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    Fix deprecation warnings in :class:`.NaturalGradient`, which now uses the
+    now use :class:`~sklearn.preprocessing.StandardScaler` to scale the data
```
Comment (Cryoris) [inline releasenotes/notes/switched-to-StandardScaler-43d24a7918e96c14.yaml:5]:
```suggestion
    :class:`~sklearn.preprocessing.StandardScaler` to scale the data
```

File: releasenotes/notes/switched-to-StandardScaler-43d24a7918e96c14.yaml (hunk 2 above)
Comment (Cryoris) [reply releasenotes/notes/switched-to-StandardScaler-43d24a7918e96c14.yaml]: Oh yep, thank you!

**Review by Cryoris — APPROVED** [review]
